node/test/Test/Pos/Client/Txp/UtilSpec.hs

```diff
@@ -0,0 +1,246 @@
+-- | Specification of Pos.Client.Txp.Util
+
+module Test.Pos.Client.Txp.UtilSpec
+       ( spec
+       ) where
+
+import           Universum
+import           Unsafe                   (unsafeFromJust)
+
+import qualified Data.ByteString          as BS
+import qualified Data.List.NonEmpty       as NE
+import qualified Data.Map                 as M
+import qualified Data.Set                 as S
+import           Formatting               (build, hex, left, sformat, shown, (%), (%.))
+import           Test.Hspec               (Spec, describe)
+import           Test.Hspec.QuickCheck    (prop)
+
+import           Pos.Client.Txp.Addresses (MonadAddresses (..))
+import           Pos.Client.Txp.Util      (TxError (..), TxOutputs, TxWithSpendings,
+                                           createMTx, createRedemptionTx)
+import           Pos.Core                 (HasCoreConstants, unsafeIntegerToCoin)
+import           Pos.Crypto               (RedeemSecretKey, SafeSigner, SecretKey,
+                                           decodeHash, fakeSigner)
+import           Pos.Txp                  (TxId, TxIn (..), TxOut (..), TxOutAux (..),
+                                           Utxo)
+import           Pos.Types                (Address)
+import           Test.Pos.Util            (giveTestsConsts, stopProperty)
+
+import           Test.Pos.Client.Txp.Mode (TxpTestMode, TxpTestProperty)
+import           Test.Pos.Client.Txp.Util (generateAddressAux, generateRedeemAddressAux,
+                                           seedSize)
+
+----------------------------------------------------------------------------
+-- Tests
+----------------------------------------------------------------------------
+
+spec :: Spec
+spec = giveTestsConsts $ describe "Client.Txp.Util" $ do
+    describe "createMTx" $ createMTxSpec
+
+createMTxSpec :: HasCoreConstants => Spec
+createMTxSpec = do
+    prop createMTxWorksWhenWeAreRichDesc createMTxWorksWhenWeAreRichSpec
+    prop stabilizationDoesNotFailDesc stabilizationDoesNotFailSpec
+    prop feeIsNonzeroDesc feeIsNonzeroSpec
+    prop manyUtxoTo1Desc manyUtxoTo1Spec
+    prop manyAddressesTo1Desc manyAddressesTo1Spec
+    prop manyAddressesToManyDesc manyAddressesToManySpec
+    prop redemptionDesc redemptionSpec
+    prop txWithRedeemOutputFailsDesc txWithRedeemOutputFailsSpec
+  where
+    createMTxWorksWhenWeAreRichDesc =
+        "Transaction is created successfully when we have 1 input with 1M coins " <>
+        "and 1 output with 1 coin"
+    stabilizationDoesNotFailDesc =
+        "The message \"Couldn't stabilize tx fee after 5 attempts\" is not thrown " <>
```

Comment:
I don't think it's a good idea to use error message instead of exception type here. I would write that `FailedToStabilize` is not thrown.

Reply:
This is a leftover from before the `TxError` refactoring. Fixed.